src/Compilers/CSharp/Test/Emit2/Semantics/PatternMatchingTests5.cs

```diff
@@ -3148,5 +3148,148 @@
                     );
             }
         }
+
+        [Fact, WorkItem(45679, "https://github.com/dotnet/roslyn/issues/45679")]
+        public void IsExpression_SwitchDispatch_Numeric()
+        {
+            var source = """
+class C
+{
+    public static bool Test(int a)
+    {
+        return (a is 1 or 2 or 3 or 4 or 5 or 6 or 7 or 8);
```

Comment:
nit: Consider adding a similar test with non-contiguous values (should emit a switch IL instruction if they are packed enough, or a binary tree of conditions if they are further apart).